File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int minDepth(TreeNode root) {
        if (root == null) return 0;
        
        int size;
        int depth = 0;
        Queue<TreeNode> queue = new LinkedList<>();
        TreeNode node;
        queue.add(root);
        while (queue.size() > 0) {
            depth++;
            size = queue.size();
            for (int i = 0; i < size; i++) {
                node = queue.remove();
                if (node.left == null && node.right == null) return depth; // leaf
                if (node.left != null) queue.add(node.left);
                if (node.right != null) queue.add(node.right);
            }
        }        
        return depth;
    }
}

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque

class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        depth = 0
        q = deque()
        q.append([root])
        while q:
            depth += 1
            level = q.popleft()
            next_level = []
            for node in level:
                if not node.left and not node.right: # leaf
                    return depth
                if node.left:
                    next_level.append(node.left)
                if node.right:
                    next_level.append(node.right)
            q.append(next_level)
                
        return depth
                